src/SFML/Window/WindowImpl.cpp

```diff
@@ -175,30 +176,50 @@ void WindowImpl::setMaximumSize(const std::optional<Vector2u>& maximumSize)
 
 
 ////////////////////////////////////////////////////////////
-Event WindowImpl::popEvent(bool block)
+Event WindowImpl::popEventBlocking(Time timeout)
 {
+    const auto startTime = std::chrono::steady_clock::now();
```

Comment:
I considered and tried using `sf::Clock` here but the code was just not as readable / understandable due to `sf::Clock` being stateful.

Reply:
What about: 

```suggestion
    static Clock clock;
    const auto startTime = clock.restart();
```

Reply:
Does not seem like an improvement to me TBH, and it introduces `static` function-object synchronization overhead.

Reply:
I argue it simplifies things, as we're not reinventing the wheel. Can you give a benchmark that this overhead is measurable in any meaningful way for event handling?

Reply:
I'd argue the opposite, `sf::Clock` is reinventing the wheel: the standard C++ way of measuring time since C++11 is `<chrono>`. 

`sf::Clock` is implemented on top of `<chrono>` to provide additional functionality that `<chrono>` does not provide which is unneeded here. We don't need a restartable/toggleable stopwatch, we just need the `::now()` function.

Following the principle to use the simplest tool to get the job done, I'm strongly in favour of sticking to `<chrono>` here.

Reply:
Resolving, feel free to re-open if you strongly feel we should use `sf::Clock`.